packages/backend-test-utils/src/next/wiring/TestBackend.ts

```diff
@@ -101,4 +127,30 @@ export async function startTestBackend<
   }
 
   await backend.start();
+
+  return backend;
 }
+
+let registered = false;
+function registerTestHooks() {
+  if (typeof afterEach !== 'function') {
+    return;
+  }
+  if (registered) {
+    return;
+  }
+  registered = true;
+
+  afterEach(async () => {
+    for (const backend of backendInstances) {
+      try {
+        await backend.stop();
+      } catch (error) {
+        console.error(`Failed to stop backend after test, ${error}`);
+      }
+    }
+    backendInstances.length = 0;
+  });
+}
+
+registerTestHooks();
```

Comment:
Hm. Given the scoping behavior of `before*` and `after*`, I wonder if this will actually not always work the way you intend it to. Feels like the only way of making things safe, is to **call** those functions at the exact time that the creation of the backend happens. I.e. you have to put their invocation directly inside `startTestBackend`. But the problem then, is that you can't do that inside an `it` - and it will be equally possible that `startTestBackend` is invoked at the root, or in a `describe`, or inside an `it`. All should ideally be made to work equally well, right? You should rewrite the test to exercise all of those cases and making sure they work and don't miss stopping or stop more than once.

Reply:
Hmm I should revise that perhaps ... I guess that what you have here works in practice 🤔 At least for afterEach, and as long as you do the creation inside an `it`. But now that the default value is `afterEach`, it'll mess up instances that were created in a `describe` if there is more than one `it`. Just highlighting that that's a pretty breaking behavior that might warrant upping the changeset level and have a special mention.

Reply:
To sum up discussion on Discord: we'll try to implement automatic `afterAll` behavior as well, so that things behave correctly when starting a backend in a describe block. Probably using `expect.getState().currentTestName`

Reply:
Turns out `expect.getState().currentTestName` doesn't work :(, it doesn't reset state after tests

We settled for stopping at the end of all tests instead.